app/attributes/attributecontroller.cpp

```diff
@@ -1276,3 +1278,82 @@ void AttributeController::onFeatureAdded( QgsFeatureId newFeatureId )
   setFeatureLayerPair( FeatureLayerPair( f, mFeatureLayerPair.layer() ) );
   emit featureIdChanged();
 }
+
+void AttributeController::renamePhotos()
+{
+  const QStringList photoNameFormat = QgsProject::instance()->entryList( QStringLiteral( "Mergin" ), QStringLiteral( "PhotoNaming/%1" ).arg( mFeatureLayerPair.layer()->id() ) );
+
```

Comment:
We could skip the entire logic here in case a user does not have anything defined to save looping over all fields.

```suggestion
if (photoNameFormat.isEmpty())
  return?
```